ideascube/library/migrations/0008_library_stock.py

```diff
@@ -23,16 +27,45 @@ def migrate_data(apps, schema_editor):
             subtitle=old_book.subtitle, publisher=old_book.publisher,
             section=old_book.section, lang=old_book.lang, cover=old_book.cover,
             tags=old_book.tags)
-        new_book.save()
+        old_specimen = None
+        new_bookspecimens = []
 
         for old_bookspecimen in old_book.specimens.all():
-            new_bookspecimen = BookSpecimen(
-                barcode=old_bookspecimen.serial, item=new_book, count=1,
-                comments=old_bookspecimen.remarks,
-                # And now the stuff which hasn't changed
-                location=old_bookspecimen.location, file=old_bookspecimen.file)
-            new_bookspecimen.save()
+            if old_bookspecimen.serial in specimens:
+                # If "old serial" equals "new barcode" then we are dealing with
+                # the same specimen
+                old_specimen = specimens.pop(old_bookspecimen.serial)
+
+            if old_specimen is not None:
+                # Either we just got old_specimen in the previous if condition,
+                # or we got it on a previous iteration
```

Comment:
> That means we can only associate a BookSpecimen with the corresponding Specimen, but we have nothing to associate a Book to its corresponding StockItem.

Yes, but this is already done at the first iteration, no need to keep it to the second iteration.
Moreover, by keeping the `old_specimen` you may associate a `new_bookspecimen` without corresponding `specimen` to the `old_specimen`. And so having two `bookspecimen` inherit from the same `specimen`